src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -607,6 +607,15 @@ private Expression<?> parseSingleExpr(boolean allowUnparsedLiteral, @Nullable Lo
 					return new SimpleLiteral<>(parsedObject, false, new UnparsedLiteral(expr));
 				}
 			}
+			if (expr.startsWith("\"") && expr.endsWith("\"")) {
+				for (ClassInfo<?> aClass : exprInfo.classes) {
+					if (!aClass.getC().isAssignableFrom(String.class))
+						continue;
```

Comment:
Literals can only be converted to other literals, so we don't really want to return a literal unless the input actually wants a string (otherwise a lot of places might break because they can't be converted)